src/expr/src/scalar/func.rs

```diff
@@ -3738,18 +3748,26 @@ fn array_create_multidim<'a>(
     Ok(datum)
 }
 
-/// Constructs a new 1D array out of an arbitrary number of scalars.
+/// Constructs a new zero or one dimensional array out of an arbitrary number of
+/// scalars.
 ///
-/// The lower bound of the array is always one. The length of the array is equal
-/// to `datums.len()`.
+/// If `datums` is empty, constructs a zero-dimensional array. Otherwise,
+/// constructs a one dimensional array whose lower bound is one and whose length
+/// is equal to `datums.len()`.
 fn array_create_scalar<'a>(
     datums: &[Datum<'a>],
     temp_storage: &'a RowArena,
 ) -> Result<Datum<'a>, EvalError> {
-    let dims = &[ArrayDimension {
+    let mut dims = &[ArrayDimension {
         lower_bound: 1,
         length: datums.len(),
-    }];
+    }][..];
```

Comment:
Why is this additional index required? Does it replace the 'static lifetime with the local one?

Reply:
Not quite! It forces a coercion from `&[ArrayDimension; 1]` to `&[ArrayDimension]`, i.e., a coercion from an array type to a slice type. Something is funky about Rust's type inference, and it can't always inject that coercion automatically.

Reply:
Ah, right of course. Thanks for the explanation!